sympy/core/equation.py

```diff
@@ -0,0 +1,359 @@
+
+Usage examples can be found in the docstrings and the demonstration Jupyter notebook ``Demonstration of algebraic_equation.py.ipynb``.
+
+J.G 28-5-20
+"""
+
+from .basic import Basic
+from .sympify import _sympify
+'''
+from sympy.simplify import simplify
+from sympy.core import Function
+from sympy.functions import *
+from sympy import functions
+from sympy import factorial
+'''
+
+#from sympy import * # Replace with more selective import when folded into SymPy
+
+class Equation(Basic):
+    """
+    This class defines an equation with a left-hand-side (lhs) and a right-hand-side (rhs) connected by an operator (e.g. $p*V = n*R*T$).
+    
+    Explanation
+    ===========
+    This class defines relations that all high school and college students would recognize as mathematical equations. 
+    At present only the "=" relation operator is recognized.
+    
+    This class is intended to allow using the mathematical tools in SymPy to rearrange equations and perform algebra
+    in a stepwise fashion. In this way more people can successfully perform algebraic rearrangements without stumbling
+    over missed details such as a negative sign.
+    
+    __Note__ that this module imports Sympy into its namespace so there is no need to import Sympy separately.
+    
+    Create an equation with the call ``Equation(lhs,rhs,relation_operator)``, where ``lhs`` and ``rhs`` are any valid Sympy
+    expression. ``relation_operator`` defaults to the string "=" if not supplied. Currently,"=" is the only valid option.
+    ``Eqn(...)`` is a synonym for ``Equation(...)``.
+    
+    Examples
+    ========
+    >>> var('a b c')
+    >>> equ(a,b/c)
+    a=b/c
+    >>> t=equ(a,b/c)
+    >>> t
+    a=b/c
+    >>> t*c
+    a*c=b
+    >>> c*t
+    a*c=b
+    >>> exp(t)
+    exp(a)=exp(b/c)
+    >>> exp(log(t))
+    a=b/c
+
+    Integration can only be performed on one side at a time.
+    >>> q=Eqn(a*c,b/c)
+    >>> integrate(q,b,side='rhs')
+    b**2/(2*c)
+    >>> integrate(q,b,side='lhs')
+    a*b*c
+
+    >>> # Make a pretty statement of integration from an equation 
+    >>> Eqn(Integral(q.lhs,b),integrate(q,b,side='rhs'))                        
+    Integral(a*c, b)=b**2/(2*c)
+    >>> # This is duplicated by the convenience function self.integ 
+    >>> q.integ(b)                                                              
+    Integral(a*c, b)=b**2/(2*c)
+
+    SymPy's solvers do not understand these equations. They expect an expression that the solver assumes = 0.
+    Thus to use the solver the equation must be rearranged so that all non-zero symbols are on one side. Then
+    just the non-zero symbolic side is passed to ``solve()``.
+    >>> t2 = t-t.rhs
+    >>> t2
+    a-b/c=0
+    >>> solve(t2.lhs,c)
+    [b/a]
+    """
+
+    def __new__(cls, lhs, rhs, relop='='):
+        if not(relop == '='):
+           raise NotImplementedError('"=" is the only relational operator presently supported in Equations.')
```

Comment:
Rather than `not(a==b)` you can use `a != b`.

If relop is always `=` then I don't think we need it as an argument. It can always be added later if needed.

Reply:
I've had inconsistent performance in python with `!=` so had been avoiding it recently. Easy enough to change. I can remove `relop` for now.

Reply:
I agree we should remove it. If we want to support other relationals they should have separate classes (with a possible superclass for common code). But I think inequalities are a lot more difficult to work with as expressions, so we should figure out Equality first.